### ISSUE TITLE
How to fix deprecation notices?

### ISSUE THREAD

**Issue by rooselle** [phake/phake#300]
PHPUnit version: 9.5.5
Phake version: 4.1.0
Symfony version: 5.3.1

I use Phake to mock my classes in a Symfony 5 app. After an upgrade from PHPUnit 4.7 to 9.3 (and I still have the problem in 9.5), without changing anything to my code, I got these deprecation notices when I run my tests:
```bash
Remaining indirect deprecation notices (3)

  1x: The "Symfony\Bridge\Twig\Mime\TemplatedEmail::__serialize()" method is considered internal. It may change without further notice. You should not extend it from "TemplatedEmail_PHAKEcbeb592e8143b41".
    1x in ContactMailerTest::testSendEmail from App\Tests\Mailer

  1x: The "Symfony\Bridge\Twig\Mime\TemplatedEmail::__unserialize()" method is considered internal. It may change without further notice. You should not extend it from "TemplatedEmail_PHAKEcbeb592e8143b41".
    1x in ContactMailerTest::testSendEmail from App\Tests\Mailer

  1x: Class "TranslatorInterface_PHAKEe9b1bfdc40995c1" should implement method "Symfony\Contracts\Translation\TranslatorInterface::getLocale()": Returns the default locale.
    1x in EmailFactoryTest::testCreateEmail from App\Tests\Mailer
```

My guess is that Phake, when mocking classes, is also mocking methods that are internal and that shouldn't be implemented; for the last deprecation, my guess is that Phake is mocking the class without implementing all methods.
Is my guess right, or at least is it a Phake problem?
If that's so, is there a fix to not be polluted by these deprecation notices (since my code *isn't* deprecated), but without changing the phpunit config to not show deprecations notices (since I want to know if my code is deprecated at some point)?

Thanks in advance!

**Comment by adoy** [comment]
Hi, 

Thanks for filling this issue. Would it be possible for you to create a small sample project on github or anywhere with an example of the issue ? 

Thanks

**Comment by rooselle** [comment]
Hi,

Yes sure! Here it is: https://github.com/rooselle/sample-phake-300
Hope it'll help!

**Comment by adoy** [comment]
Thanks for the project. :-) This is more an issue with Symfony. I created a PR there to fix your problem. You can find the link just above).
I'll close the issue for now but do not hesitate to reopen or comment on this issue if you have something new.

Thanks

**Comment by adoy** [comment]
@rooselle FYI Symfony was released. The bug fix is now included in `4.4.26`, `5.2.11` and `5.3.3`. Hope this fixes your issue.
